lib/Server.js

```diff
@@ -551,8 +551,13 @@ class Server {
         }
       });
 
-      if (this.options.client.logging) {
-        this.sockWrite([connection], 'logging', this.options.client.logging);
+      const clientLogLevel =
+        this.options.client.logging ||
+        (this.compiler.options.infrastructureLogging &&
```

Comment:
Compilers can be array here

Reply:
Ideally it should be on plugin level (and getting from the compilation), but it is require moer complex logic

Reply:
Note there is `getCompilerConfigArray()`  util.

Reply:
Yes, it should be easy to fix

Reply:
Good catch, I'll fix this tomorrow.